File: dealer-web/src/constants/tranStage.js
export const TRAN_STAGE = Object.freeze({
  QUOTATION: 'QUOTATION',
  PENDING: 'PENDING',
  APPROVED: 'APPROVED',
  PAYMENT: 'PAYMENT',
  COMPLETED: 'COMPLETED',
  LOST: 'LOST'
})

export const TRAN_STAGE_META = Object.freeze({
  [TRAN_STAGE.QUOTATION]: { type: 'info', text: '待报价' },
  [TRAN_STAGE.PENDING]: { type: 'warning', text: '待审批' },
  [TRAN_STAGE.APPROVED]: { type: 'success', text: '已审批' },
  [TRAN_STAGE.PAYMENT]: { type: 'warning', text: '待收款' },
  [TRAN_STAGE.COMPLETED]: { type: 'success', text: '已完成' },
  [TRAN_STAGE.LOST]: { type: 'danger', text: '丢失关闭' }
})

export const TRAN_STAGE_OPTIONS = Object.freeze(
  Object.entries(TRAN_STAGE_META).map(([value, meta]) => ({
    value,
    label: meta.text
  }))
)

export const getTranStageType = (stage) => TRAN_STAGE_META[stage]?.type || ''

export const getTranStageText = (stage) => TRAN_STAGE_META[stage]?.text || stage || ''

export const normalizeTranStage = (stage) => {
  if (stage && TRAN_STAGE_META[stage]) {
    return stage
  }
  return TRAN_STAGE.QUOTATION
}

File: dealer-server/src/main/java/com/autodealer/crm/enums/TranStage.java
package com.autodealer.crm.enums;

/**
 * 交易阶段。数据库存储枚举名，避免业务状态依赖历史字典数字。
 */
public enum TranStage {
    QUOTATION("待报价"),
    PENDING("待审批"),
    APPROVED("已审批"),
    PAYMENT("待收款"),
    DELIVERY("待交付"),
    COMPLETED("已完成"),
    LOST("丢失关闭"),
    CLOSED("已关闭"),
    CANCELLED("已取消");

    private final String label;

    TranStage(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}
